File: VB/WebSite/App_Code/MyObject.vb
Imports System
Imports DevExpress.Xpo

Public Class MyObject
	Inherits XPObject

	Public Sub New()
		MyBase.New()
	End Sub

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public Overrides Sub AfterConstruction()
		MyBase.AfterConstruction()
	End Sub

	Protected _Title As String
	Public Property Title() As String
		Get
			Return _Title
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("Title", _Title, value)
		End Set
	End Property

	Private _Item As Int32
	Public Property Item() As Int32
		Get
			Return _Item
		End Get
		Set(ByVal value As Int32)
			SetPropertyValue("Item", _Item, value)
		End Set
	End Property

	Private _SomeDate As DateTime
	Public Property SomeDate() As DateTime
		Get
			Return _SomeDate
		End Get
		Set(ByVal value As DateTime)
			SetPropertyValue("SomeDate", _SomeDate, value)
		End Set
	End Property
End Class

File: CS/WebSite/App_Code/MyObject.cs
using System;
using DevExpress.Xpo;

public class MyObject : XPObject {
    public MyObject()
        : base() { }

    public MyObject(Session session)
        : base(session) { }

    public override void AfterConstruction() {
        base.AfterConstruction();
    }

    protected String _Title;
    public String Title {
        get { return _Title; }
        set { SetPropertyValue<String>("Title", ref _Title, value); }
    }

    private Int32 _Item;
    public Int32 Item {
        get { return _Item; }
        set { SetPropertyValue("Item", ref _Item, value); }
    }

    private DateTime _SomeDate;
    public DateTime SomeDate {
        get { return _SomeDate; }
        set { SetPropertyValue("SomeDate", ref _SomeDate, value); }
    }
}

